### DISCUSSION TITLE
A couple of questions

### DISCUSSION THREAD

**Discussion by donlaiq** [AppFlowy-IO/AppFlowy#2956] [Technical help wanted]
Hi!

Is there any reason for having the **_.vscode_** folder inside the **_frontend_** folder instead of the root folder?
I'm asking that because I can't see the _Launch Configurations_ (described here https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/launcher-and-tasks) when I open the project from the root folder. 
So, to see them, I need to open the project from the **_frontend_** folder. 

I'm also seeing some launch configurations for Android, but I guess they don't work yet, am I right?

Best.

**Comment by jatin-web** [comment]
Hey, I am new here. Could you please help me in setting up the project? I have cloned the project and trying to run it(flutter) but there are a lot of warning in the lib folder and I tried everything but can't fix them.

**Reply by GP-commits** [reply]
Can you share the exact error you are facing?

**Comment by GP-commits** [comment]
Hi!

1. **Why `.vscode` is in `frontend`**: AppFlowy is set up as a monorepo, and the `frontend` folder is a self-contained sub-project. All the VS Code task paths and build configurations (like `cargo make` for the Rust core) expect `frontend` to be the workspace root. If you open VS Code at the repository root, these paths and tasks will break. You should open the `frontend` directory directly in VS Code to develop.

2. **Android configs**: Yes, they work! But because the app uses a Rust backend, you need to set up Android cross-compilation for Rust first. You'll need the Android NDK installed, the target added via rustup (`rustup target add aarch64-linux-android`), and `cargo-ndk` (`cargo install cargo-ndk`). Once that environment is ready, the Android launch configurations will build the Rust core and run the Flutter app on your device.

Hope this helps!
